zcash_client_backend/src/data_api.rs

```diff
@@ -1019,6 +1020,27 @@ pub trait WalletWrite: WalletRead {
         request: UnifiedAddressRequest,
     ) -> Result<Option<UnifiedAddress>, Self::Error>;
 
+    /// Generates and persists a new address for the specified account, with the specified
+    /// diversifier index.
+    ///
+    /// Returns the new address, or an error if the account identifier does not correspond to a
+    /// known account.
+    /// A conflict with an existing row in the database is considered acceptable and no error is returned.
+    /// If the diversifier index cannot produce a valid Sapling address, no sapling receiver will
+    /// be included in the returned address.
```

Comment:
I think that this method should take a `UnifiedAddressRequest` instead of having this default behavior: https://github.com/zcash/librustzcash/blob/main/zcash_keys/src/keys.rs#L421-L425

Reply:
The reason I left it out was that elsewhere, it was used to influence or check the diversifier index would work for the required receivers. In this case, the caller requires the exact index given to be inserted, so compatibility checks are moot. I didn't really need to return a UnifiedAddress from the function either. In my case I drop it on the floor. 
My concern with taking `UnifiedAddressRequest` here is that it complicates the caller, as it now must describe what it expects to be the allowed receivers (and no more than that), when that isn't necessary.

Reply:
The caller may be someone like the Brave browser, which will not be supporting the Sapling pool at all. So it _must_ be up to the caller what receivers are generated, and they must handle the error that can occur if the diversifier index does not correspond to an index that is valid for the request they provided.